src/utils/contract.ts

```diff
@@ -119,4 +117,25 @@ export default class Contract extends BaseConnection {
     });
     return await response.json();
   }
+
+  private async fetchDataViaProvider(params) {
+    switch (this.web3Provider!.providerType) {
+      case "ethers":
+        return await this.web3Provider!
+          .sendAsync('eth_call', params)
+          .then(resp => resp.json()); 
+      case "web3steam":
+        return this.web3Provider!.sendAsync({
```

Comment:
We should fall back to `send` if `sendAsync` is not implemented. This can be the case for some old adapters, like the deprecated [HttpAdapter](https://github.com/ethereum/web3.js/blob/242bfd0adfb8581efd79ef6589d782fbc1b466d8/packages/web3-providers-http/src/index.js)